identity-iota/src/did/doc/iota_document.rs

```diff
@@ -380,34 +380,38 @@ impl IotaDocument {
         unsafe { IotaVerificationMethod::new_unchecked_ref(m) })
   }
 
-  /// Adds a new [`IotaVerificationMethod`] to the DID Document.
-  pub fn insert_method(&mut self, method: IotaVerificationMethod, scope: MethodScope) -> bool {
-    self.document.insert_method(method.into(), scope)
+  /// Adds a new [`IotaVerificationMethod`] to the document in the given [`MethodScope`].
+  ///
+  /// # Errors
+  ///
+  /// Returns an error if a method with the same fragment already exists.
+  pub fn insert_method(&mut self, method: IotaVerificationMethod, scope: MethodScope) -> Result<()> {
+    Ok(self.document.insert_method(method.into(), scope)?)
```

Comment:
```suggestion
self.document.insert_method(method.into(), scope)
```

Reply:
That doesn't work, unfortunately, because the `Err` type of the `Result` in `IotaDocument` is an `identity_iota::Error` while the `Err` type of the `Result` of `CoreDocument::insert_method` is of type `identity_did::Error`.